uwrt_arm_cartesian_controller/src/uwrt_arm_cartesian_controller.cpp

```diff
@@ -0,0 +1,224 @@
+#include "uwrt_arm_cartesian_controller/uwrt_arm_cartesian_controller.hpp"
+#include "controller_interface/helpers.hpp"
+
+#include "kdl_parser/kdl_parser.hpp"
+
+#include <utility>
+#include <array>
+
+namespace uwrt_arm_cartesian_controller
+{
+    UWRTCartesianController::UWRTCartesianController() : m_cart_command_sub(nullptr), m_ik_solver(nullptr) {}
+
+    CallbackReturn UWRTCartesianController::on_init()
+    {
+        try
+        {
+            auto_declare<std::vector<std::string>>("joint_names", std::vector<std::string>());
```

Comment:
auto_declare is something defined in ros2_control packages. if u want to use that, ur gonna need to link against them and set up the deps properly